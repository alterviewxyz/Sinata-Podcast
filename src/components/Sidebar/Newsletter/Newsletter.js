// @flow
import React, { useEffect } from "react";
import styles from "./Newsletter.module.scss";

type Props = {
  copyright: string
};

const Newsletter = ({  }: Props) => {
  useEffect(() => {
    const js = `
    function ml_webform_success_1484552() {
      var $ = ml_jQuery || jQuery;
      $('.ml-subscribe-form-1484552 .ml-block-success').show();
      $('.ml-subscribe-form-1484552 .ml-block-form').hide();
    };
  `;
    const script = document.createElement("script");
    const scriptText = document.createTextNode(js);
    script.appendChild(scriptText);
    document.body.appendChild(script);

    const script2 = document.createElement("script");
    script2.src =
      "https://static.mailerlite.com/js/w/webforms.min.js?vd890ed88b3a28c805acc70e1a88fa27c";
    document.body.appendChild(script2);
  }, []);
  return (
    <section id="signup" className="gradient-gray">
      <div id="mlb2-1484552" className="ml-subscribe-form ml-subscribe-form-1484552">
        <div className="ml-vertical-align-center">
          <div className="subscribe-form ml-block-success" style={{ display: "none" }}>
            <div className="form-section">
              <p>ثبت‌نام با موفقیت انجام شد!</p>
            </div>
          </div>
          <form
            className="ml-block-form"
            action="https://landing.mailerlite.com/webforms/submit/v1b1m4"
            data-id="v1b1m4"
            data-code="v1b1m4"
            method="POST"
            target="_blank"
          >
            <span className="signup-title">خبرنامه</span>
            <div className="subscribe-form">
              <div className="form-section">
                <div className="form-group ml-field-email ml-validate-required ml-validate-email">
                  <input
                    type="email"
                    name="fields[email]"
                    className="form-control signup-text"
                    placeholder="آدرس ایمیل"
                    autoComplete="email"
                    spellCheck="false"
                    autoCapitalize="off"
                    autoCorrect="off"
                  />
                  <input type="hidden" name="ml-submit" value="1" />
                  <button type="submit" className="primary signup-button">
                    ثبت
                  </button>
                  <button disabled style={{ display: "none" }} type="button" className="loading">
                    در حال ثبت
                    {/* <img
                  src="https://static.mailerlite.com/images/rolling@2x.gif"
                  alt="loading..."
                  width="20"
                  height="20"
                  style={{ width: '20px', height: '20px' }}
                /> */}
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </section>
  );
};

export default Newsletter;
