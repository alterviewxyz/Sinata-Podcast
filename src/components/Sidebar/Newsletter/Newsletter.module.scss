@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';
