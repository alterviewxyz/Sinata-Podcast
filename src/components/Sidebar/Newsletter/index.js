// @flow
export { default } from './Newsletter';
