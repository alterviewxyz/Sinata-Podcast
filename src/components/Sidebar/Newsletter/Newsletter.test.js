// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Newsletter from './Newsletter';

describe('Newsletter', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Newsletter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
