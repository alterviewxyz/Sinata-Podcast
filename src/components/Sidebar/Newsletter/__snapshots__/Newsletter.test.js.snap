// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter renders correctly 1`] = `
<section
  className="gradient-gray"
  id="signup"
>
  <div
    className="ml-subscribe-form ml-subscribe-form-1484552"
    id="mlb2-1484552"
  >
    <div
      className="ml-vertical-align-center"
    >
      <div
        className="subscribe-form ml-block-success"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="form-section"
        >
          <p>
            ثبت‌نام با موفقیت انجام شد!
          </p>
        </div>
      </div>
      <form
        action="https://landing.mailerlite.com/webforms/submit/v1b1m4"
        className="ml-block-form"
        data-code="v1b1m4"
        data-id="v1b1m4"
        method="POST"
        target="_blank"
      >
        <span
          className="signup-title"
        >
          خبرنامه
        </span>
        <div
          className="subscribe-form"
        >
          <div
            className="form-section"
          >
            <div
              className="form-group ml-field-email ml-validate-required ml-validate-email"
            >
              <input
                autoCapitalize="off"
                autoComplete="email"
                autoCorrect="off"
                className="form-control signup-text"
                name="fields[email]"
                placeholder="آدرس ایمیل"
                spellCheck="false"
                type="email"
              />
              <input
                name="ml-submit"
                type="hidden"
                value="1"
              />
              <button
                className="primary signup-button"
                type="submit"
              >
                ثبت
              </button>
              <button
                className="loading"
                disabled={true}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="button"
              >
                در حال ثبت
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</section>
`;
